frontend/recorder/SaveScreen.tsx

```diff
@@ -111,32 +113,43 @@ export class _SaveScreen extends React.PureComponent<SaveScreenProps> {
                         className='bp3-input bp3-fill'
                         value={wavAudioUrl || ''}
                         readOnly
+                        onFocus={this.handleFocus}
                     />
                 </FormGroup>}
                 <FormGroup label="Target">
                     <HTMLSelect options={grantOptions} value={targetUrl} onChange={this.handleTargetChange}/>
                 </FormGroup>
-                <Button
-                    onClick={this.onUpload}
-                    disabled={!canUpload}
-                    intent={canUpload ? Intent.PRIMARY : Intent.NONE}
-                    icon='floppy-disk' text="Save"
-                />
-                <div>
+                <div className="encoding-status">
                     {busy ?
-                        <Spinner size={Spinner.SIZE_SMALL} />
+                        <Spinner size={Spinner.SIZE_SMALL} className="mr-2" />
                     : (step === 'done' ?
                         <Icon icon='tick' intent={Intent.SUCCESS} />
                     : null)}
 
                     {message}
                 </div>
-                {typeof progress === 'number' &&
+                {typeof progress === 'number' && 'done' !== step &&
                     <ProgressBar value={progress}/>
                 }
+                <div className="mt-4">
+                    {canUpload && <Button
+                        onClick={this.onUpload}
+                        intent={canUpload ? Intent.PRIMARY : Intent.NONE}
+                        icon='floppy-disk'
+                        className="mr-2"
+                        text={getMessage('UPLOADING_BUTTON')}
+                    />}
+                    <Button
```

Comment:
The "Cancel" button deletes the current recording if it hasn't been saved. It should be called "Delete recording" or something like that, and have a trash bin icon.

Going with [my comment](https://github.com/France-ioi/codecast/pull/188#pullrequestreview-627912449) on the pull request #188 , I think there should also be a button "Resume recording" which just closes the dialog and allows to continue the recording.